### DISCUSSION TITLE
plugin not authenticating

### DISCUSSION THREAD

**Discussion by kjustie** [homebridge-simplisafe3/homebridge-simplisafe3#401] [Q&A]
Anybody else had Simplisafe authentication issues over the last couple weeks?  My plugin stopped responding about 3 weeks ago; when it didn't resolve itself after a few days , I re-did the authentication process, and that got it to work for about a week, then it failed again. And now, re-doing authentication process (tried twice) does not resolve it, even temporarily.  Everything seems OK with the plugin during a Homebridge restart (just the usual errors about not recognizing the newer cameras).  But then the log starts showing "SSAPI request failed, request blocked (rate limit or auth failure?)" every minute and Home app shows "no response" for the Simplisafe system.   Everything is up-to-date in terms of Homebridge and the Simplisafe plug-in, no recent OS updates on the Homebridge server, and controlling the system from the Simplisafe app, itself, works fine.   Any suggestions appreciated -- thx!

**Comment by shamoon** [comment]
No issues here (and not a lot of reports recently). Sometimes it does seem your IP can get "banned" (seemingly with increasing aggressiveness) by the API servers. The plugin makes some effort to back off but another thing to try is to turn it off completely (yes, inconvenient) for ~24h and try again

**Comment by kjustie** [comment]
Thanks for the suggestion;  I've had it disabled for 2+ days, just re-enabled it now, and it's working.  hopefully that will last!  
The documentation says, for sensorRefresh, "setting this value too low will likely lead to your IP address being (temporarily) blocked by SimpliSafe."  The default is 15, but I had mine set at 30, so this would not seem likely to have been the problem, right?

**Reply by shamoon** [reply]
No agreed I dont think youre `sensorRefresh` is the issue

Yea fingers crossed...

**Comment by kjustie** [comment]
Well, it only lasted for a month -- exactly, so possibly not just a coincidence?   re-enabled plug-in 9/14 and it worked through 10/14, then died yesterday 10/15.   Full error stream after a restart:

[10/16/2023, 7:02:53 PM] [home alarm] SSAPI request failed, request blocked (rate limit or auth failure?).
[10/16/2023, 7:02:53 PM] [home alarm] Accessory refresh failed due to rate limiting or connectivity: RateLimitError: Error: getaddrinfo ENOTFOUND api.simplisafe.com
    at SimpliSafe3.request (/usr/local/lib/node_modules/homebridge-simplisafe3/simplisafe.js:228:30)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at SimpliSafe3.getLocks (/usr/local/lib/node_modules/homebridge-simplisafe3/simplisafe.js:397:30)
    at SS3Platform.discoverSimpliSafeDevices (/usr/local/lib/node_modules/homebridge-simplisafe3/index.js:265:19)

Any options other than temporarily disabling it again...and potentially having to repeat that every month?   Increase refresh interval?  (currently at 30)

**Reply by shamoon** [reply]
Certainly is odd. Not really, no. I have been running this plugin personally for years at this point (not to mention maintaining this project) and it is extremely rare that I have to re-auth. If you are indeed getting blocked it’s usually your IP so eg assuming you’re on the same network you can try the api url in a browser and it will fail there too.

It’s been a while since I looked at one of these, I don’t recall if ENOTFOUND is typical for these, that’s usually a dns issue as opposed to a server blocking you, but that might even be a quirk of how they’re doing rate limiting

**Comment by kjustie** [comment]
Hi again.  After keeping the plugin disabled for 1, then 2, then 4 days and still getting the rate limiting error, I finally turned it off for a week and that worked: no errors, of any type, on a Homebridge restart (other than the usual ones about the newer types of cameras).  But, things still don't work:  within the Home app, Simplisafe says "Updating" for a minute or so, and then eventually "No Response."  I turned on logging in the plugin and restarted.  It eventually showed

[11/6/2023, 2:00:41 PM] [home alarm] SimpliSafe credentials refresh was successful
[11/6/2023, 2:00:43 PM] [home alarm] Updated current state for SimpliSafe 3: OFF
[11/6/2023, 2:00:43 PM] [home alarm] Updated current state for SimpliSafe 3: OFF
[11/6/2023, 2:00:44 PM] [home alarm] SSAPI socket `open`
[11/6/2023, 2:00:44 PM] [home alarm] SSAPI socket `hello`
[11/6/2023, 2:00:44 PM] [home alarm] SSAPI socket `registered`
[11/6/2023, 2:00:44 PM] [home alarm] SSAPI socket `subscribed`
[11/6/2023, 2:00:44 PM] [home alarm] SimpliSafe real time events connected

...and then adding a line like this...

[11/6/2023, 2:06:02 PM] [home alarm] SSAPI socket `heartbeat`

...every minute, with the refresh rate now set at 60 (had been 30).  

No errors of any type in the entire restart process.   So are we looking at some other type of issue with my server?   I have two other plugins:  Nest, which is working fine (can control thermostats), and MyQ for garage door -- which is also not working, though with a "Forbidden API call" error, which would seem unrelated to my Simplisafe issue (and which I *think* started after Simpliesafe broke; I've not started to investigate that one yet).

Other than the occasional need to re-auth the Simplisafe login, my system has worked fine for several years, including approaching 3 years on the current hardware (MacBook Air).  There have been no OS upgrades on the server (it can't take anything beyond Big Sur) and it's up-to-date on Homebridge updates.   Nor any network or router changes.   So I'm stumped;  any reason to try reinstalling the plugin?   Other ideas/suggestions?   Thx!

**Reply by shamoon** [reply]
1. If the plugin is working but HomeKit isn't connected to it usually restarting all home hubs helps. In general this plugin (and homebridge in general) doesnt really "control" the connection to HomeKit, it just publishes the info and HomeKit communicates with homebridge.
2. If that doesnt help you could 'remove all cached accessories'. Note this will break all of your automations but they can be backed up and restored with an app such as Controller+. Still, a nuclear option.
3. The myQ plugin was broken by the company, it seems permanently, unfortunately. IDK why that dev hasn't made that more obvious on the plugin page or discord, but thats what happened (and no, has nothing to do with this). Hoping SS doesnt go that way some day.

**Comment by kjustie** [comment]
Took me a while to get back to this, and while it wasn't in your list of suggestions, I did just update node.js and homebridge, itself; not surprisingly, Simplisafe plugin is still not working.  I also (right after your last message), restarted all Home hubs (latest model AppleTV is active hub, with an older ATV and a Homepod Mini in standby) and that did not resolve the problem.

I haven't gone on to your step 2, because I belatedly discovered this:  In the Homebridge interface (website) on my server, in the Accessories tab, the SimpliSafe plugin DOES work.   But again, on my phone, in the Home app, it says "no response," and it's not an issue with my phone, as I get the same response in the Home app on my laptop and iPad.  But does the fact that it works on the server change your diagnosis of where any problem might lie?   (If not, when you suggested "Controller+"  for backup in your step 2 were you referring to the paid version of Controller for Homekit?  I don't find a "Contoller+" on iOS or Mac app stores.)

**Reply by shamoon** [reply]
This is the app I meant: https://apps.apple.com/app/id1198176727

Well yes if things are working as expected under homebridge accessories the problem is between homebridge and HomeKit, not this plugin and homebridge, if that makes sense.

Depending on how determined / patient you are, yea I would just really nuke everything and start fresh (restoring with that app once you’ve got things working). Just be prepared for real worst case where you need to manually redo everything but hopefully the app prevents that (there are some things it doesn’t back up, explained in the app)
